### PR TITLE
Add protocol detection timeout log message to protocol detection doc

### LINKED ISSUE #1480: include log message in protocol detection documentation
Protocol detection timeouts generate a log message like this:
```
INFO ThreadId(01) outbound:server{orig_dst=74.125.123.76:443}: linkerd_detect: Continuing after timeout: linkerd_proxy_http::version::Version protocol detection timed out after 10s
```

If we include (some of) that string in the protocol detection doc, it will be easier for people to find the solution!

### PR DESCRIPTION
**Pull request by dtaskai**
Fixes #1480, added the protocol detection log message to the docs for better SEO.

Signed-off-by: Dominik Táskai <dominik.taskai@leannet.eu>

### CODE REVIEW

File: linkerd.io/content/2.12/features/protocol-detection.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ decrypt the connection to observe the HTTP transactions.)
 ## Configuring protocol detection
 
 {{< note >}}
-If you are experiencing 10-second delays when establishing connections, you are
+If you receive the `protocol detection timed out after 10s` log message and
+are experiencing 10-second delays when establishing connections, you are
```
Comment (hawkw) [inline linkerd.io/content/2.12/features/protocol-detection.md:30]: this is pretty nitpicky, but I don't love the use of the word "receive" here. how about something like this:
> If your proxy logs contain messages like `protocol detection timed out after 10s`, or you are experiencing 10-second delays [...]

**Review by hawkw — CHANGES_REQUESTED** [review]

File: linkerd.io/content/2.12/features/protocol-detection.md (hunk 1 above)
Comment (dtaskai) [reply linkerd.io/content/2.12/features/protocol-detection.md:30]: It also didn't sit well with me, but now I see why :smile:, thank you @hawkw!

**Review by wmorgan — APPROVED** [review]
Thank you!

**Comment by wmorgan** [comment]
@hawkw you good with this?

**Review by hawkw — APPROVED** [review]
Looks good to me!

**Pull request merged**